tests/support/redis.tcl

```diff
@@ -247,30 +247,46 @@ proc ::redis::redis_read_null fd {
     return {}
 }
 
+proc ::redis::redis_read_bool fd {
+    set v [redis_read_line $fd]
+    if {$v == "t"} {return 1}
+    if {$v == "f"} {return 0}
+    return -code error "Bad protocol, '$v' as bool type"
+}
+
 proc ::redis::redis_read_reply {id fd} {
     if {$::redis::readraw($id)} {
         return [redis_read_line $fd]
     }
 
-    set type [read $fd 1]
-    switch -exact -- $type {
-        _ {redis_read_null $fd}
-        : -
-        + {redis_read_line $fd}
-        , {expr {double([redis_read_line $fd])}}
-        - {return -code error [redis_read_line $fd]}
-        $ {redis_bulk_read $fd}
-        > -
-        ~ -
-        * {redis_multi_bulk_read $id $fd}
-        % {redis_read_map $id $fd}
-        default {
-            if {$type eq {}} {
-                catch {close $fd}
-                set ::redis::fd($id) {}
-                return -code error "I/O error reading reply"
+    while {1} {
+        set type [read $fd 1]
+        switch -exact -- $type {
+            _ {return [redis_read_null $fd]}
+            : -
+            ( -
+            + {return [redis_read_line $fd]}
+            , {return [expr {double([redis_read_line $fd])}]}
+            # {return [redis_read_bool $fd]}
```

Comment:
My editor (and GitHub's diff) thinks this line is a comment, but it isn't. We could write this as `\#` or `"#"` here to unconfuse syntax highlighing. Both of these work (empirically tried).

Good PR btw! I like it.

Reply:
yes, this line seems to confuse everyone (you're the 4th person to comment on it, others did it in private chats).
adding quotes or slash will mess up the indentation, unless we do it for all lines.
i suppose that we don't care much now that it's merged (this file is not frequently edited).

btw, this is a perfectly valid Tcl code. the editors are all wrong.
it turns out that comments in Tcl don't work everywhere, for instance you can't comment in lists.
have you ever tried to put one of the lines in `::all_tests` in test_helper.tcl in a comment? (it doesn't work).

i vote to keep it as is... we should learn Tcl better, instead of trying to modify the code to what we think it should have done.
similar in my eyes as adding an extra pair of parenthesis in C, just because we don't know the operator precedence.

Reply:
Fine. (I'm not confused. Only the editors.)